src/cljs/ataru/virkailija/application/application_search_control.cljs

```diff
@@ -78,41 +78,66 @@
       "/lomake-editori/applications/complete"
       (str "Käsitellyt haut" (haku-count-str @complete-count))]]))
 
-(defn haku-info-link [link-href {:keys [name application-count processed]}]
-  [:a
-   {:href link-href}
-   (some #(get name %) [:fi :sv :en])
-   (str " (" application-count ")")
-   (let [unprocessed (- application-count processed)]
-     (when (pos? unprocessed)
-       [:span.application__search-control-haku-unprocessed
-        (str " " unprocessed " Käsittelemättä")]))])
+(defn haku-info-link [link-href {:keys [name haku-application-count application-count unprocessed processed]}]
+  (let [processing (- application-count unprocessed processed)]
+    [:a.application__search-control-haku-link
+     {:href link-href}
+     [:span.application__search-control-haku-title
+      (some #(get name %) [:fi :sv :en])]
+     [:span.application__search-control-haku-hl]
+     (when haku-application-count
+       [:span.application__search-control-haku-count (str haku-application-count " hakemus" (when (< 1 haku-application-count) "ta"))])
+     [:span.application-handling__count-tag.application-handling__count-tag--haku-list
+      [:span.application-handling__state-label.application-handling__state-label--unprocessed]
+      unprocessed]
+     [:span.application-handling__count-tag.application-handling__count-tag--haku-list
+      [:span.application-handling__state-label.application-handling__state-label--processing]
+      processing]
+     [:span.application-handling__count-tag.application-handling__count-tag--haku-list
+      [:span.application-handling__state-label.application-handling__state-label--processed]
+      processed]]))
 
-(defn hakukohde-list [hakukohteet]
-  [:div (map
-         (fn [hakukohde]
-           ^{:key (:oid hakukohde)}
-           [:div.application__search-control-hakukohde
-            [haku-info-link
-             (str "/lomake-editori/applications/hakukohde/" (:oid hakukohde))
-             hakukohde]])
-         hakukohteet)])
+(defn hakukohde-list [hakukohteet-opened hakukohteet]
+  [:div.application__search-control-hakukohde-container
+   (let [hakukohde-count (count hakukohteet)]
+     (if @hakukohteet-opened
+       [:div.application__search-control-hakukohteet
+        (when (and @hakukohteet-opened (pos? hakukohde-count))
```

Comment:
`@hakukohteet-opened` on tässä aina `true`. `(pos? hakukohteet-count)` sijaan voisi olla `(not-empty hakukohteet)`.